### PR TITLE
Added application flow and some input fields.

### PR DESCRIPTION
**Pull request by fjandakornid**
# Document Provider Onboarding

## What

Adding application flow and some input fields.

## Why

To have some foundation to build on.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  buildIntroductionField,
+  buildTextField,
+  Form,
+  ApplicationTypes,
+} from '@island.is/application/core'
+import { m } from './messages'
+
+export const ContactInfo: Form = buildForm({
+  id: ApplicationTypes.DOCUMENT_PROVIDER_ONBOARDING,
+  ownerId: 'TODO?',
```
Comment (arnifreyrs) [inline libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts:15]: If you rebase to master this attribute is no longer there

File: libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                }),
+                buildTextField({
+                  id: 'administrativeContact.phoneNumber',
+                  name: m.administrativeContactPhoneNumber,
+                }),
+                buildTextField({
+                  id: 'administrativeContact.address',
+                  name: m.administrativeContactAddress,
+                }),
+                buildTextField({
```
Comment (arnifreyrs) [inline libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts:80]: you can use `width: 'half'` if you want to change the width of fields

File: libs/application/templates/document-provider-onboarding/src/forms/ReviewApplication.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            buildReviewField({
+              id: 'approvedByReviewer',
+              name: 'Samþykkir þú þessa umsókn?',
+              actions: [
+                { event: 'APPROVE', name: 'Samþykkja', type: 'primary' },
+                { event: 'REJECT', name: 'Hafna', type: 'reject' },
+              ],
+            }),
+            buildTextField({
+              id: 'rejectionReason', // this is not working atm, probably since it's on the last page
```
Comment (arnifreyrs) [inline libs/application/templates/document-provider-onboarding/src/forms/ReviewApplication.ts:44]: yes, the second to last page is the "submission" page. It is a very bad hack that needs to be fixed. I just don't know exactly how the API should be....

**Review by arnifreyrs — APPROVED** [review]
nice!

**Review by hinriksteinar — APPROVED** [review]

**Review by ssh16 — APPROVED** [review]
Very nice

File: libs/application/templates/document-provider-onboarding/src/forms/ReviewApplication.ts (hunk 3 above)
Comment (fjandakornid) [reply libs/application/templates/document-provider-onboarding/src/forms/ReviewApplication.ts:44]: :+1:

File: libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts (hunk 2 above)
Comment (fjandakornid) [reply libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts:80]: :+1:

File: libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts (hunk 1 above)
Comment (fjandakornid) [reply libs/application/templates/document-provider-onboarding/src/forms/DocumentProviderApplication.ts:15]: Ahh ok... I was trying to find out what this was, but now I don't have to :)

**Pull request merged**